src/Swarm/Game/Entity.hs

```diff
@@ -448,9 +450,49 @@ entityInventory = hashedLens _entityInventory (\e x -> e {_entityInventory = x})
 -- Inventory
 ------------------------------------------------------------
 
--- | A convenient synonym to remind us when an 'Int' is supposed to
---   represent /how many/ of something we have.
-type Count = Int
+-- | A type that represent /how many/ of something we have.
+--
+-- In some challenges it is useful to have _an infinite_ amount
+-- of supplies in robots inventory as that is more visible
+-- than hidden default devices.
+data Count = Count Natural | Infinity
+  deriving (Eq, Ord, Show, Read, Generic, Data)
+
+instance ToJSON Count where
+  toJSON (Count a) = toJSON a
+  toJSON Infinity = Null
```

Comment:
Ugh, no, I do not want to have to write  `[tree, null]` in a `.yaml` file when I actually want infinite trees.

Reply:
I guess we can write `infinity` which will be passed as a string?

Reply:
Yes, I think that should work.

Reply:
:heavy_check_mark: